### PR TITLE
prompt-studio helper changes for v2 porting

### PR DESCRIPTION
**Pull request by johnyrahul**
## What

- Porting v2 changes for the below PR's
1. https://github.com/Zipstack/unstract/pull/571
2. https://github.com/Zipstack/unstract/pull/587

## Why

-

## How

-

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

-

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py [hunk 1]
```diff
@@ -780,7 +780,7 @@ def dynamic_indexer(
         vector_db = str(profile_manager.vector_store.id)
         x2text_adapter = str(profile_manager.x2text.id)
         extract_file_path: Optional[str] = None
-
+        directory, filename = os.path.split(file_path)
         if not is_summary:
             directory, filename = os.path.split(file_path)
```
Comment (ritwik-g) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py:785]: Do we need this redundant logic?

**Review by muhammad-ali-e — APPROVED** [review]
please Address @ritwik-g 's comment

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py [hunk 2]
```diff
@@ -780,7 +780,7 @@ def dynamic_indexer(
         vector_db = str(profile_manager.vector_store.id)
         x2text_adapter = str(profile_manager.x2text.id)
         extract_file_path: Optional[str] = None
-
+        directory, filename = os.path.split(file_path)
```
Comment (harini-venkataraman) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py:783]: @johnyrahul Handled this change in this PR - https://github.com/Zipstack/unstract/pull/720

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py (hunk 1 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py]: updated

**Review by hari-kuriakose — APPROVED** [review]
@johnyrahul LGTM overall

**Pull request merged**